src/uetai/callbacks/image_monitor.py

```diff
@@ -1,136 +1,162 @@
+        :type media: Dict[str, List]
         :param tag: The tag the images
         :type tag: str, optional
+        """
+        images = []
+        for idx, item in enumerate(media['images']):
+            predict = media['predictions'][idx]
+            gt = media['ground_truths'][idx]
+            if self._label_mapping is not None:
+                # TODO: check mapping valid
+                predict = self._label_mapping[predict]  # mapping label
+            images.append(wandb.Image(item, caption=f'Ground truth: {gt} Prediction: {predict}'))
 
-        .. example::
-            .. code::python
-            >>> buh buh lmao
+        self._add_image(tag=tag, images=images)
 
+    def _add_image(self, tag: str, images: Union[wandb.Image, List[wandb.Image]],) -> None:
         """
-        images = []
-        if tag is None:
-            tag = 'Media'
-        for item in batch:
-            # map label's id to name
-            label = item[0]
-            image = item[1]
-            images.append(wandb.Image(image, caption=label))
-        self.__add_image(tag=tag, images=images)
-
-    def __add_image(
-        self, tag: str, images: List or wandb.Image,
-    ) -> None:
-        """Override this method to customize the logging of Image.
+        Log image(s) to Weight & Biases dashboard.
 
         :param tag: The name of the logging panel in dashboard
         :type tag: str
-        :param images: The `wandb.Image` that going to be logging
-        :type images: List
+        :param images: The `wandb.Image` or List of images that going to be logging
+        :type images: List[wandb.Image] or wandb.Image
         """
         if wandb is None:
-            raise ImportError(
-                "To log image with `wandb`, please it install with `pip install wandb`"
-            )
+            raise ImportError("To log image with `wandb`, please it install with `pip install wandb`")
         logger = self._trainer.logger
-        logger.wandb_run.log({tag: images})
-
-    def _check_logger(self, logger: LightningLoggerBase) -> bool:
-        available = True
-        if not logger:
-            rank_zero_warn("Cannot log image because Trainer has no logger.")
-            available = False
-        elif not isinstance(logger, SummaryWriter):
-            rank_zero_warn(
-                f"{self.__class__.__name__} does not "
-                "support logging with {logger.__class__.__name__}."
-            )
-            available = False
-        else:
-            if 'wandb' not in logger.log_tool:
-                rank_zero_warn(
-                    "Current `Summary_Writer` is not running with `wandb`."
-                    "Please enable `wandb` to log image"
-                )
-        return available
+        if isinstance(logger, SummaryWriter) and 'wandb' in logger.log_tool:
+            logger.wandb_run.log({tag: images})
+
+
+# class ClassificationMonitor(ImageMonitorBase):
```

Comment:
remove any unused code

Reply:
Yes, I am testing a new specific callback for image classification.